tests/unit_tests/db_engine_specs/test_mssql.py

```diff
@@ -46,6 +47,7 @@
         ("NCHAR(10)", UnicodeText, None, GenericDataType.STRING, False),
         ("NVARCHAR(10)", UnicodeText, None, GenericDataType.STRING, False),
         ("NTEXT", UnicodeText, None, GenericDataType.STRING, False),
+        ("uniqueidentifier", GUID, None, GenericDataType.STRING, False),
```

Comment:
very minor nit: to validate caseless matching, I would maybe consider changing the test to 
```suggestion
        ("UNIQUEIDENTIFIER", GUID, None, GenericDataType.STRING, False),
```